lib/redcord/base.rb

```diff
@@ -61,4 +61,14 @@ def self.included(klass)
       attribute :updated_at, T.nilable(Time), index: true
     end
   end
+
+  sig { returns(T::Array[T.class_of(Redcord::Base)]) }
+  def self.descendants
+    descendants = []
+    ObjectSpace.each_object(Class) do |klass|
```

Comment:
```suggestion
    ObjectSpace.each_object(T::Struct) do |klass|
```
we should reduce the search space as much as possible

Reply:
Using `T::Struct` didn't seem to work. I'll try again

Reply:
Using `T::Struct` returns `[]` for descendants within this function. I think broader might be necessary.

Reply:
Update: @donaldong and I paired and we are unsure why T::Struct doesn't behave like class. I'll add a todo but otherwise keep it as `Class` for now